src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonArray.IList.cs

```diff
@@ -31,12 +31,21 @@ public void Add(JsonNode? item)
         /// </summary>
         public void Clear()
         {
-            for (int i = 0; i < List.Count; i++)
+            List<JsonNode?>? list = _list;
+
+            if (list is null)
             {
-                DetachParent(List[i]);
+                _jsonElement = null;
```

Comment:
Why does this get set to null if list is null but not if it's not null?

Reply:
It can only be not null if list is null. This is because of how the constructors and the `InitializeList` method work.